packages/desktop/views/dashboard/settings/views/advanced/BalanceFinder.svelte

```diff
@@ -1,23 +1,33 @@
 <script lang="ts">
-    import { Button, Text, ButtonSize } from 'shared/components'
+    import { Button, Text, ButtonSize, Checkbox, TextType } from 'shared/components'
     import { localize } from '@core/i18n'
     import { openPopup, PopupId } from '@auxiliary/popup'
 
+    let searchInCurrentWallet: boolean = false
+
     function onBalanceFinderClick(): void {
         openPopup({
             id: PopupId.BalanceFinder,
             props: {
                 title: localize('popups.balanceFinder.title'),
                 body: localize('popups.balanceFinder.body'),
+                searchInCurrentWallet,
             },
         })
     }
 </script>
 
-<Text type="h4" classes="mb-3">{localize('views.settings.balanceFinder.title')}</Text>
-<Text type="p" secondary classes="mb-5">{localize('views.settings.balanceFinder.description')}</Text>
-<div class="flex flex-row items-center">
-    <Button size={ButtonSize.Medium} inlineStyle="min-width: 156px;" onClick={onBalanceFinderClick}>
-        {localize('actions.findBalances')}
-    </Button>
+<Text type={TextType.h4} classes="mb-3">{localize('views.settings.balanceFinder.title')}</Text>
+<Text type={TextType.p} secondary classes="mb-5">{localize('views.settings.balanceFinder.description')}</Text>
+<div class="flex flex-col space-y-4">
+    <div class="flex flex-row items-center">
+        <Button size={ButtonSize.Medium} inlineStyle="min-width: 156px;" onClick={onBalanceFinderClick}>
+            {localize('actions.findBalances')}
+        </Button>
+    </div>
+    <Checkbox
```

Comment:
I would probably move this checkbox inside the popup itself, and each time the checkbox changes value, we could reset the search algorithm
I would also hide the wallet llist and the total balance if the user selects to search only on the selected wallet